x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/persistence/DataFrameAnalyticsConfigProvider.java

```diff
@@ -112,11 +134,62 @@ public void get(String id, ActionListener<DataFrameAnalyticsConfig> listener) {
      * @param ids a comma separated list of single IDs and/or wildcards
      */
     public void getMultiple(String ids, boolean allowNoMatch, ActionListener<List<DataFrameAnalyticsConfig>> listener) {
+        LOGGER.info("Getting multiple analytics [{}]", ids);
         GetDataFrameAnalyticsAction.Request request = new GetDataFrameAnalyticsAction.Request();
         request.setPageParams(new PageParams(0, MAX_CONFIGS_SIZE));
         request.setResourceId(ids);
         request.setAllowNoResources(allowNoMatch);
         executeAsyncWithOrigin(client, ML_ORIGIN, GetDataFrameAnalyticsAction.INSTANCE, request, ActionListener.wrap(
             response -> listener.onResponse(response.getResources().results()), listener::onFailure));
     }
+
+    /**
+     * Unlike {@link #getMultiple(String, boolean, ActionListener)} this method tries to get the configs that match jobs with tasks.
+     * It expects concrete ids and it does not throw if there is no config for a given id.
+     */
+    public void getConfigsForJobsWithTasksLeniently(Set<String> jobsWithTask, ActionListener<List<DataFrameAnalyticsConfig>> listener) {
```

Comment:
I tried to put this method in `MlMemoryTracker` directly, but it made the tests in `MlMemoryTrackerTests` really hard to get working due to having to mock the client call.